src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -272,27 +272,16 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     * Fetches voting data for the blocks and stores any relevant
     * result into the appropriate database table
     */
-  private[this] def processVotesForBlocks(blocks: List[TezosTypes.Block]): Future[Option[Int]] = {
-    import cats.syntax.traverse._
-    import cats.syntax.foldable._
-    import cats.instances.list._
-    import cats.instances.option._
-    import cats.instances.int._
-    import slickeffect.implicits._
-    import slick.jdbc.PostgresProfile.api._
-
+  private[this] def processVotesForBlocks(blocks: List[TezosTypes.Block]): Future[Option[Int]] =
     tezosNodeOperator.getVotingDetails(blocks).flatMap {
       case (proposals, bakersBlocks, ballotsBlocks) =>
         //this is a nested list, each block with many baker rolls
-        val writeBakers = bakersBlocks.traverse {
-          case (block, bakersRolls) => TezosDb.writeVotingRolls(bakersRolls, block)
-        }
+        val writeBakers = TezosDb.writeVotingRolls(bakersBlocks)
 
-        val combinedVoteWrites = writeBakers.map(_.combineAll.map(_ + proposals.size + ballotsBlocks.size))
+        val combinedVoteWrites = writeBakers.map(_.map(_ + proposals.size + ballotsBlocks.size))
 
-        db.run(combinedVoteWrites.transactionally)
+        db.run(combinedVoteWrites)
```

Comment:
@piotrkosecki What would happen if the db fails in the middle?
Do you think the `Tables.Rolls ++= bakers` operation is actually atomic at the db level? Or it might lose only part of  the vote writes?

Reply:
According to the comment in the slick if any part of the batch fails it throws an exception - it uses JDB~C batch update feature.